### PR TITLE
fixed bug where delimiters in a table name would result in incorrect …

### PR DESCRIPTION
**Pull request by msgroi**

### CODE REVIEW

File: src/test/java/com/salesforce/dynamodbv2/mt/util/CachingAmazonDynamoDbStreamsTest.java [hunk 1]
```diff
@@ -910,7 +910,7 @@ void testTrimHorizonChange() {
      * problem is that caching TRIM_HORIZON makes this situation more likely.
      */
     @Test
-    @Ignore
+    @Disabled
     void testTrimHorizonMoves() {
```
Comment (msgroi) [inline src/test/java/com/salesforce/dynamodbv2/mt/util/CachingAmazonDynamoDbStreamsTest.java]: Unrelated change.  This test was being ignored in an odd way.

File: src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldPrefixFunction.java [hunk 2]
```diff
@@ -18,32 +22,26 @@
  */
 class FieldPrefixFunction {
 
-    private final String delimiter;
+    private final CompositeStrings compositeStrings;
 
     FieldPrefixFunction(String delimiter) {
-        this.delimiter = delimiter;
+        Preconditions.checkArgument(delimiter.length() == 1, "only single character delimiters are supported");
```
Comment (horwitzj) [inline src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldPrefixFunction.java]: "single character" -> "single-character" (i.e., add hyphen) (also add in parallel test, of course)

File: src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldPrefixFunction.java [hunk 3]
```diff
@@ -18,32 +22,26 @@
  */
 class FieldPrefixFunction {
 
-    private final String delimiter;
+    private final CompositeStrings compositeStrings;
 
     FieldPrefixFunction(String delimiter) {
```
Comment (horwitzj) [inline src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldPrefixFunction.java]: any reason not to just make `String delimiter` into `char delimiter`?

File: src/test/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldPrefixFunctionTest.java [hunk 4]
```diff
@@ -11,32 +11,45 @@
...
  */
 class FieldPrefixFunctionTest {
 
-    private static final FieldPrefixFunction SUT = new FieldPrefixFunction(".");
+    private static final FieldPrefixFunction SUT = new FieldPrefixFunction('.');
+
+    static Stream<Object[]> data() {
+        return Arrays.stream(new Object[][] {
+            { "ctx", "table", "value", "ctx.table.value" },
+            { "ctx", "com.salesforce.zero.someObject", "value", "ctx.com\\.salesforce\\.zero\\.someObject.value" }
```
[the remark this answers is not in the document]
Comment (msgroi) [reply src/test/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldPrefixFunctionTest.java]: like this ... 4336720a5933d1ad3cb8de7fa09874ca2f9b28a9?
